metsim/metsim.py

```diff
@@ -148,12 +148,84 @@ def __init__(self, params: dict):
         logger.setLevel(self.params['verbose'])
         ch.setLevel(self.params['verbose'])
         logger.addHandler(ch)
+        logger.info("read_domain")
         self.domain = io.read_domain(self.params)
+        self._normalize_times()
+        logger.info("read_met_data")
         self.met_data = io.read_met_data(self.params, self.domain)
+        self._validate_force_times(force_times=self.met_data['time'])
+        logger.info("read_state")
         self.state = io.read_state(self.params, self.domain)
         self.met_data['elev'] = self.domain['elev']
         self.met_data['lat'] = self.domain['lat']
+        logger.info("_aggregate_state")
         self._aggregate_state()
+        logger.info("load_inputs")
+        self.load_inputs()
```

Comment:
I'm wary of having `self.load_inputs()` after `self._aggregate_state()`.  The whole reason we were doing the `load` and `close` in the IO routines was to avoid the situation where the state and input forcings are looking at the same file and undefined behavior occurred.  It might be possible to reason through and prove that this situation won't occur with this setup, but I think it's easier to side with caution on this one.